docs/available-filters/record-encryption/aws-kms/setup.adoc

```diff
@@ -0,0 +1,108 @@
+:aws:  https://docs.aws.amazon.com/
+
+= AWS Key Management Service
+
+The filter is able to integrate with {aws}/kms/latest/developerguide/overview.html[AWS Key Management Service].
+
+Follow the instructions below to prepare your AWS account.  You'll need a privileged AWS user
+that is capable of creating users and policies to perform the set-up.
+
+[#_establish_an_aliasing_convention_for_keys_within_aws_kms]
+== Establish an aliasing convention for keys within AWS KMS
+
+The filter references KEKs within AWS via an {aws}/kms/latest/developerguide/alias-about.html[AWS key alias].
+
+It is necessary to establish a naming convention for the alias names.This will allow the keys used by the
+filter to be kept separate from any keys used by other systems. This document assumes that the naming convention
+will be to prefix the alias used by the filter with the word `KEK_`. If a different naming convention is used, adapt
+the instructions accordingly.
+
+== Administrator Actor
+
+To use the filter, there must be an administrative identity established within AWS IAM.  This user, which is likely to be a human,
+has the responsibility to manage keys and aliases within AWS KMS for use by the filter.
+
+Use AWS IAM to create a `kroxylicious-admin` user. Attach the policies `AWSKeyManagementServicePowerUser`] and `IAMUserChangePassword`
+to the user. You may wish to attach policy `AWSCloudShellFullAccess` so the Administrator can use the AWS CloudShell to
+use CLI to manage the KEKs. Grant the user access to the Console.
+
+If using the CLI, the following commands can be used to establish the Administrator user.  This example illustrates using the
+user-name `kroxylicious-admin`.  The choice of name is not significant.  If a different user-name is used, adapt the
+instructions accordingly.
+
+[source,shell]
+----
+ADMIN=kroxylicious-admin
+INITIAL_PASSWORD=$(tr -dc 'A-Za-z0-9!?%=' < /dev/urandom | head -c 10)
+CONSOLE_URL=https://$(aws sts get-caller-identity --query Account --output text).signin.aws.amazon.com/console
+aws iam create-user --user-name ${ADMIN}
+aws iam attach-user-policy --user-name ${ADMIN} --policy-arn arn:aws:iam::aws:policy/AWSKeyManagementServicePowerUser
+aws iam attach-user-policy --user-name ${ADMIN} --policy-arn arn:aws:iam::aws:policy/IAMUserChangePassword
+aws iam attach-user-policy --user-name ${ADMIN} --policy-arn arn:aws:iam::aws:policy/AWSCloudShellFullAccess
+aws iam create-login-profile --user-name ${ADMIN} --password ${INITIAL_PASSWORD} --password-reset-required
+echo Now log in at ${CONSOLE_URL}  with user name ${ADMIN} password ${INITIAL_PASSWORD} and change the password.
+----
```

Comment:
I'm a bit confused by this section. I think what your trying to say is there are two roles/actors. An administrative identity, which manages KEKs and a Filter identity which manages DEKs. 

By naming this user `kroxylicious-admin` it sort of implies the proxy needs an admin user for operations but it then goes onto suggest policies that the proxy doesn't need. I'm not sure why this document needs or wants to document the Admin user. It could just as easily be the users ROOT account, no? I think the distinction between the admin and filter users is worth noting but I don't think we should be saying much if anything about the admin user setup. We should probably call out the permissions it needs and maybe suggest `arn:aws:iam::aws:policy/AWSKeyManagementServicePowerUser` as a means of achieving that. 

I guess we could use the above as an example of a dedicated admin user but I'm not sure we want to go there.

Reply:
> By naming this user kroxylicious-admin it sort of implies the proxy needs an admin user for operations but it then goes onto suggest policies that the proxy doesn't need.

My intent was to say there needs to be someone/something separate that has the role of managing the KEKs.  How the user achieves that is really up to them.   I didn't really want to suggest root because a) least privilege b) it might be (wrongly) interpreted as 'hey Kroxylicious needs root privileges to operate it'.   My intent was to use this section to demonstrate a way to setup a minimal admin user to do the job.  I didn't want it to be seen as prescriptive. 

Maybe I've got the balance wrong.  I'm happy to rework (but separately - please raise an issue).  We've got the same pattern in the Vault docs, so it would need to be addressed too.